File: developUtils/textFormatting.js
import fs from 'node:fs';
import path from 'node:path';
import pangu from 'pangu';

const processJson = (data) => {
  let result = Array.isArray(data) ? [] : {};

  if (typeof data === 'string') {
    // 如果是字符串，调用 A 函数并返回结果
    return pangu.spacing(data);
  } else if (Array.isArray(data)) {
    // 如果是数组，递归处理数组中的每个元素
    for (let i = 0; i < data.length; i++) {
      result.push(processJson(data[i]));
    }
    return result;
  } else if (typeof data === 'object' && data !== null) {
    // 如果是对象，递归处理对象的每一个键值对
    for (let key in data) {
      if (Object.prototype.hasOwnProperty.call(data, key)) {
        result[key] = processJson(data[key]);
      }
    }
    return result;
  } else {
    // 如果是其他类型，直接返回
    return data;
  }
};

// 从 JSON 文件读取数据并处理
const processJsonFile = (inputFilePath, outputFilePath) => {
  // 处理 JSON 数据
  fs.readFile(inputFilePath, 'utf8', (err, data) => {
    if (err) {
      console.error('读取文件失败:', err);
      return;
    }

    // 解析 JSON 数据
    let jsonData;
    try {
      jsonData = JSON.parse(data);
    } catch (err) {
      console.error('JSON 解析失败:', err);
      return;
    }

    // 处理 JSON 数据
    const processedData = processJson(jsonData);

    // 写回到文件
    fs.writeFile(
      outputFilePath,
      JSON.stringify(processedData, null, 2),
      (err) => {
        if (err) {
          console.error('写入文件失败:', err);
          return;
        }
        console.log('文件处理并成功保存!');
      }
    );
  });
};

const processFilesInDirectory = (directoryPath) => {
  fs.readdir(directoryPath, (err, files) => {
    if (err) {
      console.error('读取目录失败:', err);
      return;
    }

    files.forEach((file) => {
      const filePath = path.join(directoryPath, file);

      // 获取文件状态
      fs.stat(filePath, (err, stats) => {
        if (err) {
          console.error(`获取文件状态失败: ${filePath}`, err);
          return;
        }

        // 如果是目录，则递归调用
        if (stats.isDirectory()) {
          processFilesInDirectory(filePath);
        } else if (stats.isFile() && filePath.endsWith('.json')) {
          // 如果是 JSON 文件，则处理它
          const outputFilePath = filePath.replace('.json', '_processed.json');
          processJsonFile(filePath, outputFilePath);
        }
      });
    });
  });
};
processFilesInDirectory('../locales/zh-cn');

File: developUtils/textFormatting.py
import os
import json
import sys
import pangu

# 假设 A 是你提供的处理字符串的函数


def A(ss):
    # 这里是你自己的逻辑
    return pangu.spacing_text(ss)  # 过一边盘古

# 递归处理 JSON 数据中的字符串


def process_json(data):
    if isinstance(data, str):
        return A(data)
    elif isinstance(data, list):
        return [process_json(item) for item in data]
    elif isinstance(data, dict):
        return {key: process_json(value) for key, value in data.items()}
    else:
        return data

# 读取和处理 JSON 文件


def process_json_file(file_path, output_file_path):
    try:
        # 读取 JSON 文件
        with open(file_path, 'r', encoding='utf-8') as file:
            data = json.load(file)

        # 处理 JSON 数据
        processed_data = process_json(data)

        # 写回到文件
        with open(output_file_path, 'w', encoding='utf-8') as file:
            json.dump(processed_data, file, ensure_ascii=False, indent=2)

        print(f'文件处理并成功保存: {output_file_path}')

    except Exception as e:
        print(f'处理文件 {file_path} 时出错: {e}')

# 递归读取目录并处理所有 JSON 文件


def process_files_in_directory(directory_path):
    for root, dirs, files in os.walk(directory_path):
        for file in files:
            if file.endswith('.json'):
                file_path = os.path.join(root, file)
                process_json_file(file_path, file_path)


def main(directory_path='../locales'):
    process_files_in_directory(directory_path)


if __name__ == '__main__':
    if len(sys.argv) > 2:
        print("用法: python script.py <顶层目录 default=../locales> ")
        sys.exit(1)
    if len(sys.argv) > 1:
        main(sys.argv[1])
    else:
        main()
